drivers/hud_path_report/app/models/hud_path_report/generators/fy2024/base.rb

```diff
@@ -348,11 +353,21 @@ def initialize(generator = nil, report = nil, options: {})
         group(:DateProvided).pluck(:DateProvided, Arel.sql(array_agg(s_t[:TypeProvided]).to_sql)).to_h
     end
 
+    private def cmh_service_provided(enrollment)
+      enrollment.services.path_service.between(start_date: @report.start_date, end_date: @report.end_date).
+        pluck(Arel.sql(s_t[:TypeProvided].to_sql)).include?(4)
+    end
+
     private def path_referrals(enrollment)
       enrollment.services.path_referral.between(start_date: @report.start_date, end_date: @report.end_date).
         group(:DateProvided).pluck(:DateProvided, Arel.sql(array_agg(sql_array(s_t[:TypeProvided], s_t[:ReferralOutcome])).to_sql)).to_h
     end
 
+    private def cmh_referral_provided_and_attained(enrollment)
+      enrollment.services.path_referral.between(start_date: @report.start_date, end_date: @report.end_date).
+        pluck(Arel.sql(sql_array(s_t[:TypeProvided], s_t[:ReferralOutcome]))).include?([1, 1])
```

Comment:
Does this need the `sql_array`?  I _think_ the following would give you back an array of ruby arrays:
```suggestion
        pluck(:TypeProvided, :ReferralOutcome).include?([1, 1])
```